libcontainer/seccomp/seccomp_linux.go

```diff
@@ -134,6 +141,21 @@ func InitSeccomp(config *configs.Seccomp) (int, error) {
 	return seccompFd, nil
 }
 
+func syscallIsAlwaysAllowed(syscall *configs.Syscall) bool {
+	// We cannot guarantee if the syscall is fully allowed when additional args
+	// are provided, therefore we exclude that case.
+	if syscall == nil || len(syscall.Args) > 0 {
+		return false
+	}
+
+	switch syscall.Action {
+	case configs.Allow, configs.Trace, configs.Log:
+		return true
+	default:
+		return false
+	}
+}
```

Comment:
@rata I changed the logic to include an additional cehck for the args. Do you think this is feasible?

Reply:
As I mentioned [here][link] we might need to check the architecture matches?

I mean, what we need to know if the write done by runc will be allowed or not. For that to happen, if args is nil is simpelr to see (no need to really parse the language) but the arch  needs to match too. I don't know what would be the right thing to do if arch SCMP_ARCH_X86 is used but this is a SCMP_ARCH_X86_64 host (probably write will not match, as probably runc will be running in 64 bits and therefore not match that rule, even if the container started later uses x86 binaries?).

I think with those restrictions, we can say it will be used and be safe. But I'd like others to comment, in case I'm missing something.

cc @alban @cyphar @kolyshkin 


[link]: https://github.com/opencontainers/runc/pull/3549#issuecomment-1206317116